File: LeetCode Solutions/July 2022 LeetCoding Challenge/Max Area of Island.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {
        int maxArea = 0;
        for(int i=0; i<grid.length; i++){
           for(int j=0; j<grid[0].length; j++){
               if(grid[i][j]==1){
                   int islandArea = helper(grid, i, j);
                   if(islandArea>maxArea){
                       maxArea = islandArea;
                   }
               }
           }
        }
        return maxArea;
    }
    public int helper(int[][] grid, int i, int j){
        if(i>=0 && i<grid.length && j>=0 && j<grid[0].length && grid[i][j] == 1){
            grid[i][j] = 0;
            return(helper(grid, i+1, j) + helper(grid, i-1, j) + helper(grid, i, j-1) + helper(grid, i, j+1)) + 1;
        }
        return 0;
    }
}

File: LeetCode Solutions/Algorithm I/Day 7 Breadth-First Search - Depth-First Search/Max Area of Island.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        maxArea = 0
        def helper(row, column):
            grid[row][column] = 0
            area = 1
            for y, x in ((row-1, column), (row+1, column), (row, column+1), (row, column-1)):
                if(0 <= y < len(grid) and 0 <= x < len(grid[0]) and grid[y][x] == 1):
                    area += helper(y, x)
            return area
        for row in range(len(grid)):
            for column in range(len(grid[0])):
                if(grid[row][column] == 1):
                    maxArea = max(maxArea, helper(row, column))
        return maxArea